### PR TITLE
JAVAFIXTURE-6 Add Specimen for Temporal interface and java.time.package

### PR DESCRIPTION
**Pull request by akutschera**
That should cover most of the cases where JavaFIxture is used.
There are still some exotic classes like java.time.chrono.Era which need to be covered

### CODE REVIEW

File: src/main/java/com/github/nylle/javafixture/SpecimenFactory.java [hunk 1]
```diff
@@ -43,10 +51,38 @@ public SpecimenFactory(Context context) {
...
+            return new TimeSpecimen<>( type );
+        }
+        if(Temporal.class.isAssignableFrom(type)) {
+            return new TemporalSpecimen<>( type, context );
+        }
 
         return new ObjectSpecimen<>(type, context, this);
     }
 
+    private static boolean isTimeSpecimen( Class type ) {
```
Comment (Nylle) [inline src/main/java/com/github/nylle/javafixture/SpecimenFactory.java]: Maybe move that method to `ReflectionHelper`?

Comment (Nylle) [inline src/main/java/com/github/nylle/javafixture/SpecimenFactory.java]: Maybe move that method to `ReflectionHelper`? Then it needs a different name though...

File: src/main/java/com/github/nylle/javafixture/specimen/TimeSpecimen.java [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+
+        if ( !isTimeClass( type ) ) {
+            throw new IllegalArgumentException( "type: " + type.getName() );
+        }
+
+        this.type = type;
+        this.random = new Random();
+    }
+
+    private static boolean isTimeClass( Class type ) {
```
Comment (Nylle) [inline src/main/java/com/github/nylle/javafixture/specimen/TimeSpecimen.java]: If `isTimeSpecimen` has been moved to `ReflectionHelper`, could it be re-used here instead?

File: src/test/java/com/github/nylle/javafixture/specimen/TemporalSpecimenTest.java [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+    }
+
+    @Test
+    void throwsExceptionForUnknownType() {
+        assertThatThrownBy( () -> new TemporalSpecimen<>( Mockito.mock( Temporal.class ).getClass(), context ).create() )
+                .isInstanceOf( SpecimenException.class )
+                .hasMessageContaining( "Unsupported type:" );
+    }
+
+    @TestWithCases
```
Comment (Nylle) [inline src/test/java/com/github/nylle/javafixture/specimen/TemporalSpecimenTest.java]: Love that!!

File: src/main/java/com/github/nylle/javafixture/SpecimenFactory.java (hunk 1 above)
Comment (akutschera) [reply src/main/java/com/github/nylle/javafixture/SpecimenFactory.java]: done

File: src/main/java/com/github/nylle/javafixture/specimen/TimeSpecimen.java (hunk 2 above)
Comment (akutschera) [reply src/main/java/com/github/nylle/javafixture/specimen/TimeSpecimen.java]: yes. Thanks for spotting the duplicate code

**Pull request merged**